### PR TITLE
SQS Receiver - Update Disposal

### PR DESCRIPTION
**Pull request by pjwilliams2**
## Description
1. Updates receiver disposal to incorporate a cancellation token for propagation to the underlying SQS client
2. Updates receiver to terminate processing when cancellation has been requested by disposal
3. Prevent multiple disposals

### CODE REVIEW

File: RockLib.Messaging.SQS/SQSReceiver.cs [hunk 1]
```diff
@@ -366,38 +377,45 @@ private async Task HandleAsync(Message message)
...
+                async () =>
+                    await SQSClient.DeleteMessageAsync(
+                        new DeleteMessageRequest
+                        {
+                            QueueUrl = QueueUrl?.OriginalString,
+                            ReceiptHandle = receiptHandle
+                        },
+                        cancellationToken)
+                    .ConfigureAwait(false),
+                _consumerToken.Token);
```
Comment (pjwilliams2) [inline RockLib.Messaging.SQS/SQSReceiver.cs:391]: Real change here - passing of token. All others formatting.

**Review by kristycurrier — APPROVED** [review]

**Review by kristycurrier — CHANGES_REQUESTED** [review]
Step for unit tests is failing. Looks like:
`warning : The .NET SDK has newer analyzers with version '7.0.0' than what version '6.0.0' of 'Microsoft.CodeAnalysis.NetAnalyzers' package provides. Update or remove this package reference. `

**Comment by NeilSchiefer** [comment]
> Should there be tests around this change?

If we can.  I've found it really difficult to mock things and test the way I want to because of the way these Messaging libraries were designed.

**Comment by pjwilliams2** [comment]
> Should there be tests around this change?

Yes, updated to have some.

**Review by NeilSchiefer — APPROVED** [review]

**Review by kristycurrier — APPROVED** [review]

**Review by eslutz — APPROVED** [review]

**Pull request merged**